docs/kube-api-custom-ocp-versions.md

```diff
@@ -0,0 +1,94 @@
+# Custom OpenShift Versions (Dynamic z-stream Support)
+
+As part of [Hive Integration](https://github.com/openshift/assisted-service/blob/master/docs/Kube-API.md), a means to dynamically (without restarting the service) update an OCP version is required.
+This is mainly useful for disconnected environments that use the kube-api support. E.g. updating the release image to a new z-stream version. 
+
+In order to enable this functionality, the following components are utilized:
+* [ClusterImageSet CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#openshift-version)
+* [OPENSHIFT_VERSIONS environment variable](https://github.com/openshift/assisted-service/blob/master/hack/generate.sh#L61)
+
+### [ClusterImageSet](https://github.com/openshift/hive/blob/master/apis/hive/v1/clusterimageset_types.go)
+
+The ClusterImageSet is used for referencing to a OpenShift release image.
+So available versions are represented in an Hive cluster by defined ClusterImageSet resources.
+To use a specific release image, it should be defined in the [ClusterDeployment CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment) in either of the following manners:
+* As a reference to the ClusterImageSet in ```spec.provisioning.imageSetRef``` property.
+* Explicitly as a URL in ```spec.provisioning.releaseImage``` property.
+
+An example of a ClusterImageSet:
+```
+apiVersion: hive.openshift.io/v1
+kind: ClusterImageSet
+metadata:
+  name: openshift-v4.7.0
+spec:
+  releaseImage: quay.io/openshift-release-dev/ocp-release:4.7.0-x86_64
+```
+
+### [OPENSHIFT_VERSIONS](https://github.com/openshift/assisted-service/blob/master/Makefile#L36)
+
+The OPENSHIFT_VERSIONS environment variable is a map of OpenShift version (y-stream) to related properties as release version/image and RHCOS version/image: see [default_ocp_versions.json](https://github.com/openshift/assisted-service/blob/master/default_ocp_versions.json)
+
+Defining this var prior to starting the service would override all versions supported by the service. I.e. by default, [default_ocp_versions.json](https://github.com/openshift/assisted-service/blob/master/default_ocp_versions.json) is used for setting the supported versions. Defining OPENSHIFT_VERSIONS with a similar json format should override these.
```

Comment:
need to explain that only minimal data is required. `rhcos_image` `rhcos_version`